File: dailyques/Q2943.py
#Q2943.py
class Solution:
    def maximizeSquareHoleArea(self, n, m, h, v):
        def f(a):
            a.sort()
            best = cur = 1
            for i in range(1, len(a)):
                if a[i] == a[i-1] + 1:
                    cur += 1
                else:
                    cur = 1
                best = max(best, cur)
            return best + 1

        side = min(f(h), f(v))
        return side * side

File: dailyques/Q2943.cpp
//Q2943.cpp
class Solution {
public:
    int maximizeSquareHoleArea(int n, int m, vector<int>& h, vector<int>& v) {
        auto f = [&](vector<int>& a){
            sort(a.begin(), a.end());
            int best = 1, cur = 1;
            for(int i=1;i<a.size();i++){
                if(a[i] == a[i-1] + 1) cur++;
                else cur = 1;
                best = max(best, cur);
            }
            return best + 1;
        };
        int hs = f(h), vs = f(v);
        int side = min(hs, vs);
        return side * side;
    }
};
